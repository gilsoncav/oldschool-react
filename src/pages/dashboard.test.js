import Dashboard from './dashboard';

jest.mock('shortid', () => ({
  ...jest.requireActual('shortid'),
  generate: jest
    .fn()
    .mockReturnValueOnce('id1')
    .mockReturnValueOnce('id2')
    .mockReturnValueOnce('id3')
    .mockReturnValueOnce('id4'),
}));

describe('Dashboard page component', () => {
  it('renders correctly if game list is loading', () => {
    const wrapper = shallow(<Dashboard />);
    wrapper.setState({
      isLoading: true,
    });
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly if has finished loading', () => {
    const wrapper = shallow(<Dashboard />);
    wrapper.setState({
      gamesData: {
        total: 2000,
        list: [{ total: 1000, platform: 'SNES' }, { total: 1000, platform: 'PSX' }],
      },
      favsData: {
        total: 1500,
        list: [{ total: 500, platform: 'SNES' }, { total: 1000, platform: 'PSX' }],
      },
      isLoading: false,
    });
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });
});
