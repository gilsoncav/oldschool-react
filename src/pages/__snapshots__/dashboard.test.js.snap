// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard page component renders correctly if game list is empty 1`] = `
<LayoutAuth>
  <h1
    className="text-center text-primary"
    style={
      Object {
        "margin": "15px",
      }
    }
  >
    There are no games available, please try again later...
  </h1>
</LayoutAuth>
`;

exports[`Dashboard page component renders correctly if game list is loading 1`] = `
<LayoutAuth>
  <Row
    className="text-center"
    gutter={0}
  >
    <Spin
      size="large"
      spinning={true}
      style={
        Object {
          "marginTop": "25px",
        }
      }
      tip="Loading games..."
      wrapperClassName=""
    />
  </Row>
</LayoutAuth>
`;

exports[`Dashboard page component renders correctly if game list is not empty 1`] = `
<LayoutAuth>
  <div
    style={
      Object {
        "height": "100vh",
        "overflow": "auto",
      }
    }
  >
    <Row
      gutter={24}
      style={
        Object {
          "padding": "30px 15px",
        }
      }
    >
      <QueueAnim
        animConfig={null}
        animatingClassName={
          Array [
            "queue-anim-entering",
            "queue-anim-leaving",
          ]
        }
        appear={true}
        component="div"
        componentProps={Object {}}
        delay={0}
        duration={700}
        ease="easeOutQuart"
        forcedReplay={false}
        interval={100}
        leaveReverse={false}
        onEnd={[Function]}
        type="bottom"
      >
        <Col
          key="0"
          span={6}
          style={
            Object {
              "height": "470px",
              "marginBottom": "20px",
              "marginTop": "20px",
              "maxHeight": "470px",
              "minHeight": "470px",
            }
          }
        >
          <GameCard
            game={
              Object {
                "Name": "Super Mario Kart",
                "Overview": "The best kart game in the world, y' now",
                "Platform": "Super Nintendo Entertainment System",
                "VideoURL": null,
              }
            }
            onClick={[Function]}
          />
        </Col>
        <Col
          key="1"
          span={6}
          style={
            Object {
              "height": "470px",
              "marginBottom": "20px",
              "marginTop": "20px",
              "maxHeight": "470px",
              "minHeight": "470px",
            }
          }
        >
          <GameCard
            game={
              Object {
                "Name": "Super Mario Kart",
                "Overview": "The best kart game in the world, y' now",
                "Platform": "Super Nintendo Entertainment System",
                "VideoURL": null,
              }
            }
            onClick={[Function]}
          />
        </Col>
        <Col
          key="2"
          span={6}
          style={
            Object {
              "height": "470px",
              "marginBottom": "20px",
              "marginTop": "20px",
              "maxHeight": "470px",
              "minHeight": "470px",
            }
          }
        >
          <GameCard
            game={
              Object {
                "Name": "Super Mario Kart",
                "Overview": "The best kart game in the world, y' now",
                "Platform": "Super Nintendo Entertainment System",
                "VideoURL": null,
              }
            }
            onClick={[Function]}
          />
        </Col>
      </QueueAnim>
    </Row>
  </div>
</LayoutAuth>
`;
